### ISSUE TITLE
Backtesting does not fetch ticker_interval from the strategy

### ISSUE THREAD

**Issue by hroff-1902** [freqtrade/freqtrade#2099]
Remove ticker_interval from the config, run backtesting with a list of strategies:

```
$ freqtrade -c user_data/configs/test-config.json backtesting --strategy-list TestStrategy3 TestStrategy4
2019-08-05 23:13:51,762 - freqtrade.loggers - INFO - Verbosity set to 0
2019-08-05 23:13:51,763 - freqtrade.configuration.configuration - INFO - Dry run is enabled
2019-08-05 23:13:51,763 - freqtrade.configuration.configuration - INFO - Using DB: "sqlite:///test.sqlite"
2019-08-05 23:13:51,763 - freqtrade.configuration.check_exchange - INFO - Checking exchange...
2019-08-05 23:13:51,763 - freqtrade.configuration.check_exchange - INFO - Exchange "binance" is officially supported by the Freqtrade development team.
2019-08-05 23:13:51,763 - freqtrade.configuration.configuration - INFO - Using max_open_trades: 10 ...
2019-08-05 23:13:51,763 - freqtrade.configuration.configuration - INFO - Using data directory: user_data/data/binance ...
2019-08-05 23:13:51,764 - freqtrade.configuration.configuration - INFO - Using strategy list of 2 Strategies
2019-08-05 23:13:51,764 - freqtrade.configuration.configuration - INFO - Storing backtest results to user_data/backtest_data/backtest-result.json ...
2019-08-05 23:13:51,764 - freqtrade.optimize - INFO - Starting freqtrade in Backtesting mode
2019-08-05 23:13:51,764 - freqtrade.exchange.exchange - INFO - Instance is running with dry_run enabled
2019-08-05 23:13:51,764 - freqtrade.exchange.exchange - INFO - Applying additional ccxt config: {'enableRateLimit': True}
2019-08-05 23:13:51,768 - freqtrade.exchange.exchange - INFO - Applying additional ccxt config: {'enableRateLimit': False}
2019-08-05 23:13:51,786 - freqtrade.exchange.exchange - INFO - Using Exchange "Binance"
2019-08-05 23:13:52,578 - freqtrade.resolvers.exchange_resolver - INFO - Using resolved exchange 'Binance'...
2019-08-05 23:13:52,583 - freqtrade.resolvers.iresolver - INFO - Using resolved strategy TestStrategy3 from '/home/user/freqtrade-wrk/github-hroff-1902/freqtrade/user_data/strategies/test_strategy3.py'...
2019-08-05 23:13:52,584 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'process_only_new_candles' with value in config file: True.
2019-08-05 23:13:52,584 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'order_types' with value in config file: {'buy': 'limit', 'sell': 'limit', 'stoploss': 'market', 'stoploss_on_exchange': False, 'stoploss_on_exchange_interval': 60}.
2019-08-05 23:13:52,584 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'stake_currency' with value in config file: USDT.
2019-08-05 23:13:52,584 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'stake_amount' with value in config file: 100.
2019-08-05 23:13:52,584 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'use_sell_signal' with value in config file: True.
2019-08-05 23:13:52,584 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'sell_profit_only' with value in config file: True.
2019-08-05 23:13:52,584 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'ignore_roi_if_buy_signal' with value in config file: False.
2019-08-05 23:13:52,584 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using minimal_roi: {'40': 0.0, '30': 0.01, '20': 0.02, '0': 0.04}
2019-08-05 23:13:52,584 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using ticker_interval: 1h
2019-08-05 23:13:52,584 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stoploss: -0.1
2019-08-05 23:13:52,584 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop: False
2019-08-05 23:13:52,585 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop_positive: 0.01
2019-08-05 23:13:52,585 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop_positive_offset: 0.0
2019-08-05 23:13:52,585 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_only_offset_is_reached: False
2019-08-05 23:13:52,585 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using process_only_new_candles: True
2019-08-05 23:13:52,585 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using order_types: {'buy': 'limit', 'sell': 'limit', 'stoploss': 'market', 'stoploss_on_exchange': False, 'stoploss_on_exchange_interval': 60}
2019-08-05 23:13:52,585 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using order_time_in_force: {'buy': 'gtc', 'sell': 'gtc'}
2019-08-05 23:13:52,585 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stake_currency: USDT
2019-08-05 23:13:52,585 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stake_amount: 100
2019-08-05 23:13:52,585 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using use_sell_signal: True
2019-08-05 23:13:52,585 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using sell_profit_only: True
2019-08-05 23:13:52,585 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using ignore_roi_if_buy_signal: False
2019-08-05 23:13:52,589 - freqtrade.resolvers.iresolver - INFO - Using resolved strategy TestStrategy4 from '/home/user/freqtrade-wrk/github-hroff-1902/freqtrade/user_data/strategies/test_strategy4.py'...
2019-08-05 23:13:52,590 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'process_only_new_candles' with value in config file: True.
2019-08-05 23:13:52,590 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'order_types' with value in config file: {'buy': 'limit', 'sell': 'limit', 'stoploss': 'market', 'stoploss_on_exchange': False, 'stoploss_on_exchange_interval': 60}.
2019-08-05 23:13:52,590 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'stake_currency' with value in config file: USDT.
2019-08-05 23:13:52,590 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'stake_amount' with value in config file: 100.
2019-08-05 23:13:52,590 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'use_sell_signal' with value in config file: True.
2019-08-05 23:13:52,590 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'sell_profit_only' with value in config file: True.
2019-08-05 23:13:52,590 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'ignore_roi_if_buy_signal' with value in config file: False.
2019-08-05 23:13:52,590 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using minimal_roi: {'40': 0.0, '30': 0.01, '20': 0.02, '0': 0.04}
2019-08-05 23:13:52,590 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using ticker_interval: 1d
2019-08-05 23:13:52,590 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stoploss: -0.1
2019-08-05 23:13:52,590 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop: False
2019-08-05 23:13:52,591 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop_positive_offset: 0.0
2019-08-05 23:13:52,591 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_only_offset_is_reached: False
2019-08-05 23:13:52,591 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using process_only_new_candles: True
2019-08-05 23:13:52,591 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using order_types: {'buy': 'limit', 'sell': 'limit', 'stoploss': 'market', 'stoploss_on_exchange': False, 'stoploss_on_exchange_interval': 60}
2019-08-05 23:13:52,591 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using order_time_in_force: {'buy': 'gtc', 'sell': 'gtc'}
2019-08-05 23:13:52,591 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stake_currency: USDT
2019-08-05 23:13:52,591 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stake_amount: 100
2019-08-05 23:13:52,591 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using use_sell_signal: True
2019-08-05 23:13:52,591 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using sell_profit_only: True
2019-08-05 23:13:52,591 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using ignore_roi_if_buy_signal: False
2019-08-05 23:13:52,591 - freqtrade - ERROR - Fatal exception!
Traceback (most recent call last):
  File "/home/user/freqtrade-wrk/github-hroff-1902/freqtrade/freqtrade/main.py", line 43, in main
    args.func(args)
  File "/home/user/freqtrade-wrk/github-hroff-1902/freqtrade/freqtrade/optimize/__init__.py", line 56, in start_backtesting
    backtesting = Backtesting(config)
  File "/home/user/freqtrade-wrk/github-hroff-1902/freqtrade/freqtrade/optimize/backtesting.py", line 84, in __init__
    self._set_strategy(self.strategylist[0])
  File "/home/user/freqtrade-wrk/github-hroff-1902/freqtrade/freqtrade/optimize/backtesting.py", line 93, in _set_strategy
    self.ticker_interval_mins = timeframe_to_minutes(self.ticker_interval)
  File "/home/user/freqtrade-wrk/github-hroff-1902/freqtrade/freqtrade/exchange/exchange.py", line 785, in timeframe_to_minutes
    return ccxt.Exchange.parse_timeframe(ticker_interval) // 60
  File "/home/user/freqtrade-wrk/github-hroff-1902/freqtrade/.env/lib/python3.6/site-packages/ccxt/base/exchange.py", line 1446, in parse_timeframe
    amount = int(timeframe[0:-1])
TypeError: 'NoneType' object is not subscriptable
```

It's unclear for me, what should be the behavior of backtesting when strategies define different timeframes?
* Check it at startup and quit with DependencyException, for example
* Require ticker_interval to be set always in the config and/or cli option.
* Same but only when run with a list of strategies.
* Process backtesting on diff. timeframes, loading multiple set of data.
* Something else?

**Comment by xmatthias** [comment]
It is intentional, since ticker_interval can be defined in the strategy - and assuming each strategy uses a different ticker-interval, it's never logical which interval is used in such a case.

I guess it should not fail the way it does, but gracefully instead.
Other than that - it's documented in the "strategy_list" docstring:

>Please note that ticker-interval needs to be set either in config or via command line. 

Side note: it also fails with the same error for single strategies if ticker_interval is not set anywhere (not in config,not in strategy, not on cli).

**Comment by xmatthias** [comment]
closed with #2101
